src/main/java/org/alfresco/repo/web/scripts/servlet/RemoteUserAuthenticatorFactory.java

```diff
@@ -97,9 +138,52 @@ public RemoteUserAuthenticator(WebScriptServletRequest req, WebScriptServletResp
         public boolean authenticate(RequiredAuthentication required, boolean isGuest)
         {
             boolean authenticated = false;
-            
+
+            boolean useTimeoutForAdminAccessingAdminConsole = false;
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Authenticate level required: " + required + " is guest: " + isGuest);
+            }
+
+            if (isAlwaysAllowBasicAuthForAdminConsole())
+            {
+                useTimeoutForAdminAccessingAdminConsole =
+                    RequiredAuthentication.admin.equals(required) && !isGuest && isAdminConsoleWebScript(servletReq.getServiceMatch().getWebScript());
+
+                if (logger.isDebugEnabled())
+                {
+                    logger.debug("Should ensure that the admins can login with basic auth: " + useTimeoutForAdminAccessingAdminConsole);
+                }
+
+                if (useTimeoutForAdminAccessingAdminConsole && isBasicAuthHeaderPresentForAdmin())
+                {
+                    // return REST call, after a timeout/basic auth challenge
+                    if (logger.isDebugEnabled())
+                    {
+                        logger.debug("An Admin Console request has come in with Basic Auth headers present for an admin user.");
+                    }
+                    authenticated = super.authenticate(required, isGuest);
+                    // we return here to prompt for another password, in case it was not entered correctly
+                    // this also means, that once the admin basic auth header is present,
+                    // the authentication chain will not be used for the admin console
+                    return authenticated;
+                }
+            }
             // retrieve the remote user if configured and available - authenticate that user directly
-            final String userId = getRemoteUser();
+            final String userId;
```

Comment:
if the feature is disabled then useTimeoutForAdminAccessingAdminConsole will remain false and be passed to the "userId = getRemoteUserWithTimeout(false);"
and the first thing that method does is:
` 
if (!useTimeout)
            {
                return getRemoteUser();
            }
`
so it acts just as before.
The only way exception or new code paths can happen is if the parameter to this method is "true"
and that only happens if:
1. the feature is enabled
2. it is an AdminConsole or AdminConsoleHelper webscript
3. is "admin" level required
4. basic auth header is not present for the "admin" authority

Reply:
How about moving this catch block inside the getRemoteUserWithTimeout, because it is dealing with the timeout anyways. And then move
```
            if (!useTimeout)
            {
                return getRemoteUser();
            }
```
to else block of this "if (isAlwaysAllowBasicAuthForAdminConsole())" clause.
I hope it will reduce the amount of logic a bit. Otherwise we will need a comment in the source code similar to what you describe above.